### PR TITLE
Task/la 2 module scss

### LINKED ISSUE #2: Replace css with scss modules

### LINKED ISSUE #3: Add stylelint
configurations:
- https://github.com/B1naryStudio/bsa-website/blob/development/.stylelintrc.yml
- https://github.com/B1naryStudio/bsa-website/blob/development/design/.stylelintrc.yml

### LINKED ISSUE #8: Refactor details component
There are a lot of unnecessary styles for details tag, I think we can get rid of them and it won't impact the ui. There are also replacements for details tag with .details class made on inputs, we need to get rid of them and use details tag instead

### PR DESCRIPTION
**Pull request by kulak91**

### CODE REVIEW

**Comment by kulak91** [comment]
Includes #2 , #3 (partially), #8

File: .eslintrc.yml [hunk 1]
```diff
@@ -49,9 +49,7 @@ rules:
   '@typescript-eslint/explicit-function-return-type':
     - error
     - allowTypedFunctionExpressions: true
-  '@typescript-eslint/quotes':
-    - error
-    - single
+  '@typescript-eslint/quotes': ['error', 'single', { avoidEscape: true }]
```
Comment (liza-veis) [inline .eslintrc.yml]:
```suggestion
  '@typescript-eslint/quotes':
    - error
    - single
    - avoidEscape: true
```

File: .stylelintrc.yml [hunk 2]
```diff
@@ -0,0 +1,282 @@
+
+
```
Comment (liza-veis) [inline .stylelintrc.yml]:
```suggestion
```

File: src/components/common/accordion/accordion.tsx [hunk 3]
```diff
@@ -1,19 +1,22 @@
...
   const [heading, content] = children;
-  const classNames = classnames('accordion', className);
 
   return (
-    <details className={classNames}>
-      <summary className="accordion-header">{heading}</summary>
-      <div className="accordion-body">{content}</div>
+    <details
+      className={`${styles.accordion} ${
+        grading ? styles.gradingHometask : null
```
Comment (liza-veis) [inline src/components/common/accordion/accordion.tsx]: What is this class for, I would prefer to give it more abstract name

File: src/components/common/avatar/avatar.tsx [hunk 4]
```diff
@@ -2,17 +2,19 @@ import React from 'react';
...
-  className?: string;
+  size?: 'small' | 'medium';
 };
 
-const Avatar: React.FC<Props> = ({ name, className }) => {
+const Avatar: React.FC<Props> = ({ name, size = 'medium' }) => {
   const Avatar = nameToImage[name];
   return (
-    <div className="avatar-wrapper">
+    <div className={`${styles.avatarWrapper} ${styles[size]}`}>
```
Comment (liza-veis) [inline src/components/common/avatar/avatar.tsx]: let's use classNames for this, then in future will replace it with clsx

File: src/components/common/avatar/avatar.tsx [hunk 5]
```diff
@@ -2,17 +2,19 @@ import React from 'react';
...
       <object
-        className={`avatar-object ${className}`}
+        className={styles.avatarObject}
```
Comment (liza-veis) [inline src/components/common/avatar/avatar.tsx]:
```suggestion
        className={styles.avatar}
```

File: src/components/common/details-card/details-card.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import * as styles from './details-card.module.scss';
+
+type Props = {
+  children: [React.ReactNode, React.ReactNode];
+  type?: 'reversed';
```
Comment (liza-veis) [inline src/components/common/details-card/details-card.tsx]:
```suggestion
  isReversed?: boolean
```

File: src/components/common/details-card/details-card.tsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+const DetailsCard: React.FC<Props> = ({ children, type = '' }) => {
+  const [header, body] = children;
+  const reversed = type === 'reversed' ? styles.reversed : '';
+  return (
+    <details className={styles.testimonialsDetails}>
+      <summary className={`${styles.testimonialsSummary} ${reversed}`}>
+        <span className="details-card__header">{header}</span>
+      </summary>
+
+      <div className={`${styles.testimonialsDescription} ${reversed}`}>
```
Comment (liza-veis) [inline src/components/common/details-card/details-card.tsx]:
```suggestion
    <details className={styles.wrapper}>
      <summary className={`${styles.summary} ${reversed}`}>
        <span className="details-card__header">{header}</span>
      </summary>

      <div className={`${styles.content} ${reversed}`}>
```

File: src/components/common/flute/flute.tsx [hunk 8]
```diff
@@ -1,17 +1,13 @@
...
   const audio = useRef<HTMLAudioElement | null>(null);
   const play = (): Promise<void> | undefined => audio?.current?.play();
   const pause = (): void => audio?.current?.pause();
   return (
-    <div
-      className="message__scene message__scene-flute"
-      onMouseEnter={play}
-      onMouseLeave={pause}
-    >
+    <div className={styles.sceneFlute} onMouseEnter={play} onMouseLeave={pause}>
```
Comment (liza-veis) [inline src/components/common/flute/flute.tsx]: let's name class like just flute or wrapper

File: src/components/common/intro-blue-bubble/intro-blue-bubble.tsx [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+import { Author } from '~/types';
+import { ContactIcon } from '~/enums';
+
+import * as styles from './intro-blue-bubble.module.scss';
+import { nameToIcon } from './utils/name-to-icon';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+type Contacts = {
+  ref: string;
+  className?: 'darkBlueIcon' | 'blueIcon';
```
Comment (liza-veis) [inline src/components/common/intro-blue-bubble/intro-blue-bubble.tsx]:
```suggestion
  iconColor?: 'darkBlue' | 'blue';
```

File: src/components/common/intro-blue-bubble/intro-blue-bubble.tsx [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+  title: string;
+  iconName: ContactIcon;
+};
+
+type IntroBlueBubbleProps = {
```
Comment (liza-veis) [inline src/components/common/intro-blue-bubble/intro-blue-bubble.tsx:17]:
```suggestion
type Props = {
```

File: src/components/common/timeline-of-contents/timeline-of-contents.tsx [hunk 11]
```diff
@@ -15,27 +17,26 @@ type Props = {
...
 }) => (
-  <div id="table-of-contents" className={`table-of-contents ${headerClass}`}>
+  <div id="table-of-contents" className={styles.timelineWrapper}>
     {headerTitle && (
       <h2>
         <strong>{headerTitle}</strong>
       </h2>
     )}
-    <ul className="step" {...rest}>
+    <ul className={`${styles.step} timeline`} {...rest}>
```
Comment (liza-veis) [inline src/components/common/timeline-of-contents/timeline-of-contents.tsx]: use classnames lib

File: src/components/common/timeline-of-contents/timeline-of-contents.tsx [hunk 12]
```diff
@@ -1,5 +1,7 @@
 import React from 'react';
-import './styles.css';
+// import './styles.css';
```
Comment (liza-veis) [inline src/components/common/timeline-of-contents/timeline-of-contents.tsx]:
```suggestion
```

File: src/components/common/video-wrapper/video-wrapper.tsx [hunk 13]
```diff
@@ -15,21 +14,22 @@ const VideoWrapper: React.FC<VideoWrapperProps> = ({
   duration = '00:00',
 }) => {
   return (
-    <div className="video-wrapper">
-      <Accordion>
-        <React.Fragment>
+    <div className={styles.videoWrapper}>
+      <details className={styles.videoDetails}>
+        <summary className={styles.videoHeader}>
```
Comment (liza-veis) [inline src/components/common/video-wrapper/video-wrapper.tsx]: let's use wrapper, details, summary or header, content for classnames

File: src/layouts/lecture/lecture.tsx [hunk 14]
```diff
@@ -4,7 +4,10 @@ import { isDateInThePast } from '~/utils';
 import { LecturesContext } from '~/types';
 import { Footer, Header } from '~/partials';
 
+import '~/assets/stylesheets/styles.scss';
+
 import './lecture.scss';
+// import * as styles from './lecture.module.scss';
```
Comment (liza-veis) [inline src/layouts/lecture/lecture.tsx]:
```suggestion
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: src/components/common/intro-blue-bubble/intro-blue-bubble.tsx (hunk 10 above)
Comment (kulak91) [reply src/components/common/intro-blue-bubble/intro-blue-bubble.tsx:17]: It's being used in export/import

File: src/pages/lectures/modern-java/index.mdx [hunk 15]
```diff
@@ -124,168 +81,133 @@ _Java_ - це мова програмування загального приз
...
+    <JavaStreamSnippet />
+
+    Часто використовувані операції стрімів:
+
+    - Intermediate
     - sorted
     - filter
     - map
-  - Terminal
+    - Terminal
```
Comment (liza-veis) [inline src/pages/lectures/modern-java/index.mdx:155]: Should this space be removed?

**Comment by liza-veis** [comment]
Good job 👍

File: src/pages/lectures/modern-java/index.mdx (hunk 15 above)
Comment (kulak91) [reply src/pages/lectures/modern-java/index.mdx:155]: Made it beautful. What's wrong with the straight line? 
image

Comment (kulak91) [reply src/pages/lectures/modern-java/index.mdx:155]: Good catch. Fixed

**Review by liza-veis — APPROVED** [review]

**Pull request merged**